sql/index/pilosa/lookup.go

```diff
@@ -241,6 +244,9 @@ func (l *filteredLookup) values(p sql.Partition) (*pilosa.Row, error) {
 
 		row = intersect(row, r)
 	}
+	if err := l.index.Close(); err != nil {
```

Comment:
There are several cases where it can return an error and do not reach this line. Does it matter if `l.index` is not closed?

Reply:
Yes, we have to close it when something happen (and it's not because of pilosa, but more because we have reference counting in `concurrentPilosaIndex`).
But we have to close it asap (after first loop, so we cannot defer here)

Reply:
@jfontan - check it out